### DISCUSSION TITLE
Input with '.' in name always triggers validation error in FormRequest

### DISCUSSION THREAD

**Discussion by simon-debruijn** [laravel/framework#48630] [Q&A] [answered]
Hi all,

I'm writing a unit test for a class that maps a `Illuminate\Foundation\Http\FormRequest` into a domain specific class (e.g. Contact).

This is a simplified example:
```php
final class UpdateContactMapper
{
    public static function map(UpdateContactRequest $request, string $id): Contact {
        return new Contact(
            Uuid::fromString($id),
            $request->input('contact.name')
        );
    }
}
```

It is declared in the Laravel docs that if you want to test something regarding requests, you probably should use something like `$this->post(...`. 

I would like to **only** unit test if a request of a specific type gets mapped correctly to a specific domain object.
No request lifecycle needed.

To test if a mapper does its job correctly I write a test formed like this:
```php
$request = new UpdateContactRequest();
$request->merge([
    'contact.name' => 'John'
]);
$id = '2e43a655-e24d-4de0-93f9-5e33b0c1f8ac';

$actual = UpdateContactMapper::map($request, $id);

$expected = new Contact(Uuid::fromString($id), 'John');

$this->assertEquals($expected, $actual);
```

After a lot of debugging it looks like Laravel doesn't handle `.` well in names of input values.

I've made a reproduction project where you can test it for yourself:
https://github.com/simon-debruijn/reproduction-formrequest

I've added a few tests comparing the use of a name `contactName` and `contact.name`. The ones with `contactName` succeed, the ones with `contact.name` don't.
https://github.com/simon-debruijn/reproduction-formrequest/blob/main/tests/Http/Mappers/UpdateContactMapperTest.php#L15

The request validation keeps complaining about missing the 'contact.name' field.

Shouldn't I use a field name including a `.`?
Is there something else I'm totally missing?

Thanks in advance,

**Comment by henzeb** [comment, marked as answer]
I am not sure, but maybe it has something to do with the dot notation which you can use to get a path to a certain parameter? 

it might expect something similar to:

```php
[
    'contact'=> [
        'name' => 'a name'
    ]
];
```

**Reply by simon-debruijn** [reply]
All tests succeed when using the nested name in contact as input 👍🏻
